kani-driver/src/call_cargo.rs

```diff
@@ -211,55 +211,49 @@ impl VerificationTarget {
-            {
-                // Lib targets.
-                let unsupported_types = target
-                    .kind
-                    .iter()
-                    .filter_map(|kind| {
-                        let kind_str = kind.as_str();
-                        matches!(kind_str,
-                            CRATE_TYPE_CDYLIB | CRATE_TYPE_DYLIB | CRATE_TYPE_STATICLIB |
-                            CRATE_TYPE_PROC_MACRO
-                        ).then_some(kind_str)
-                    })
-                    .collect::<Vec<_>>();
-                if unsupported_types.is_empty() {
-                    Some(VerificationTarget::Lib)
-                } else {
-                    println!(
-                        "warning: Skipped verification of `{}` due to unsupported crate-type: `{}`.",
-                        target.name,
-                        unsupported_types.join("`, `")
-                    );
-                    None
+        let (mut supported_lib, mut unsupported_lib) = (false, false);
+        for kind in &target.kind {
+            match kind.as_str() {
+                CRATE_TYPE_BIN => {
+                    // Binary targets.
+                    verification_targets.push(VerificationTarget::Bin(target.name.clone()));
+                }
+                CRATE_TYPE_LIB | CRATE_TYPE_RLIB | CRATE_TYPE_CDYLIB | CRATE_TYPE_DYLIB
+                | CRATE_TYPE_STATICLIB => {
+                    supported_lib = true;
+                }
+                CRATE_TYPE_PROC_MACRO => {
+                    unsupported_lib = true;
+                    ignored_unsupported.push(target.name.as_str());
+                }
+                CRATE_TYPE_TEST => {
+                    // Test target.
+                    if args.tests {
+                        verification_targets.push(VerificationTarget::Test(target.name.clone()));
+                    } else {
+                        ignored_tests.push(target.name.as_str());
+                    }
                 }
-            } else if target.kind.contains(&String::from(CRATE_TYPE_TEST)) {
-                // Test target.
-                if args.tests {
-                    Some(VerificationTarget::Test(target.name.clone()))
-                } else {
-                    ignored_tests.push(target.name.as_str());
-                    None
+                _ => {
+                    ignored_unsupported.push(target.name.as_str());
                 }
-            } else {
-                ignored_unsupported.push(target.name.as_str());
-                None
             }
-        })
-        .collect();
+        }
+        match (supported_lib, unsupported_lib) {
+            (true, true) => println!(
+                "warning: Skipped verification of `{}` due to unsupported crate-type: \
+                        `proc-macro`.",
+                target.name,
+            ),
+            (true, false) => verification_targets.push(VerificationTarget::Lib),
+            (_, _) => {}
```

Comment:
What happens in the case of `(false, true)`?

Reply:
For `(false, true)`, if `verbose`, we will print that the current target was skipped due to unsupported type. Otherwise, we will silently ignore the target.

Reply:
Note that the behavior hasn't changed here and it is still covered by this test: https://github.com/model-checking/kani/tree/main/tests/cargo-ui/unsupported-lib-types/proc-macro